hooks: enable polkit for unconfined processes

alfonsosanchezbeato opened the pull request:
polkit is needed in some cases for communication between services shipped in the base, for instance when systemd-networkd talks to systemd-hostnamed to set the hostname. Enable to fix these use cases. Note that this does not help yet if we want to use it from a snap.

valentindavid reviewed (commented):
Two comments so far.

* xml-core is installed. I do not think it makes sense to keep it. It believe it is only useful for sgml docs. I am guessing it is a dependency of the post install script of some of the packages.
* `/usr/lib/tmpfiles.d/polkitd.conf` points to `/etc/polkit-1/rules.d` and `/var/lib/polkit-1`. We should add those in writable-paths. The first one is probably not needed now. But I am guessing the second should be.

valentindavid commented:
>     * `/usr/lib/tmpfiles.d/polkitd.conf` points to `/etc/polkit-1/rules.d` and `/var/lib/polkit-1`. We should add those in writable-paths. The first one is probably not needed now. But I am guessing the second should be.

Nevermind for  `/var/lib/polkit-1`, it seems was useful for localauthority for 3rd party packages. And it has been removed in recent version of polkit. I do not think we need it, and I do not think we every will.

alfonsosanchezbeato commented:
> Two comments so far.
> 
> * xml-core is installed. I do not think it makes sense to keep it. It believe it is only useful for sgml docs. I am guessing it is a dependency of the post install script of some of the packages.

It is used to generate xml catalog files, which I'm not sure if polkit xml parser will need. I have removed the executables provided by xml-core, as we definitely don't need them after installation of packages has happened.

> * `/usr/lib/tmpfiles.d/polkitd.conf` points to `/etc/polkit-1/rules.d` and `/var/lib/polkit-1`. We should add those in writable-paths. The first one is probably not needed now. But I am guessing the second should be.

Goot point, I have added the one under /etc as it would be something we will eventually want for snaps or for workarounds.

alfonsosanchezbeato commented:
@valentindavid thanks for your review, I've addressed your comments now
